src/components/core/text/text.tsx

```diff
@@ -1,11 +1,84 @@
+import React from 'react';
 import type * as Stitches from '@stitches/react';
 import { styled } from 'Styles/stitches.config';
 import { modifyVariantsForStory } from 'Styles/type-utils';
+import useTheme from '@core/theme-context/use-theme';
 
-const Text = styled('p', {
-    color: '$greyLight700',
+export const type_array: TTextProps['type'][] = [
+    'hero',
+    'heading-1',
+    'heading-2',
+    'heading-3',
+    'subtitle-1',
+    'subtitle-2',
+    'paragraph-1',
+    'paragraph-2',
+    'small',
+    'extra-small',
+];
+
+export const color_array: TTextProps['color'][] = [
+    'prominent',
+    'general',
+    'less-prominent',
+    'disabled',
+    'active',
+    'hover',
+    'secondary',
+    'primary',
+];
+
+export const align_array: TTextProps['align'][] = ['left', 'center', 'right'];
 
+type TTextProps = {
+    align: 'left' | 'center' | 'right';
+    bold: boolean;
+    children?: string | JSX.Element;
+    color: 'prominent' | 'general' | 'less-prominent' | 'disabled' | 'active' | 'hover' | 'secondary' | 'primary';
+    type:
+        | 'hero'
+        | 'heading-1'
+        | 'heading-2'
+        | 'heading-3'
+        | 'subtitle-1'
+        | 'subtitle-2'
+        | 'paragraph-1'
+        | 'paragraph-2'
+        | 'small'
+        | 'extra-small';
```

Comment:
You can actually reuse the `color_array` & `type_array` for the `color` and `type` type declaration

Reply:
Sorted.